D/Diamond/build_tarballs.jl

```diff
@@ -0,0 +1,45 @@
+using BinaryBuilder, Pkg
+
+name = "Diamond"
+version = v"2.1.6"
+
+# url = "https://github.com/bbuchfink/diamond"
+# description = "Accelerated BLAST-compatible local sequence aligner"
+
+# TODO
+# - WITH_ZSTD (fails to find the library)
+# - WITH_AVX512
+# - build failures
+#   - x86_64-w64-mingw32-cxx11
+#     error during compilation: unknown identifier __cpuidex
+
+sources = [
+    # v2.1.6
+    GitSource("https://github.com/bbuchfink/diamond/",
+              "0540f86d3d5965cf9a6ef8871068999ef61211dc";
+              unpack_target="diamond"),
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir/diamond*
+mkdir build && cd build
+cmake .. \
+    -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=RELEASE
+make -j${nproc}
+make install
+install_license ../LICENSE
+"""
+
+platforms = supported_platforms(exclude = p -> Sys.iswindows(p) || nbits(p) == 32)
+platforms = expand_cxxstring_abis(platforms; skip = p -> Sys.isfreebsd(p) || (Sys.isapple(p) && arch(p) == "aarch64"))
```

Comment:
Sorry?  The [cxx03](https://buildkite.com/julialang/yggdrasil/builds/1583#0186f998-0c02-4309-83f8-31bae9433c5d/637-1952) and [cxx11](https://buildkite.com/julialang/yggdrasil/builds/1583#0186f998-0c05-4212-b8c3-8cbf577075b9/626-1941) builds of x86_64 darwin have ***identical*** git tree hash, and the log doesn't say that the built executables are locked to a specific C++ string ABI, which would be highly suspicious since we should be building with Clang, and if that wasn't the case something would be very wrong.

Reply:
I'll have to admit that I just copied the `expand_cxxstring_abis` line from the last `build_tarballs.jl` script I had, is this `expand_cxxstring_abis` line wrong in general? Any way to easily check this from my side?

Reply:
You need to expand if and only if the audit tells you so.  Otherwise, don't.